src/averageAges.js

```diff
@@ -59,6 +99,32 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+
+  let withMother = people.map((child) => {
+    const mother = people.find((person) => person.name === child.mother);
+    const motherBorn = mother ? mother.born : null;
+
+    return {
+      ...child,
+      motherInfo: motherBorn,
+    };
+  });
+
+  if (onlyWithSon) {
+    withMother = withMother.filter((man) => {
+      return man.sex === 'm';
```

Comment:
Consider using variables with meaningful names instead of magic strings and numbers